src/qilib/utils/storage/mongo.py

```diff
@@ -112,20 +112,23 @@ def _get_root(self) -> ObjectId:
         else:
             return self._collection.insert_one({'tag': ''}).inserted_id
 
-    def _retrieve_nodes_by_tag(self, tag: TagType, parent: ObjectId) -> TagType:
+    def _retrieve_nodes_by_tag(self, tag: TagType, parent: ObjectId, document_limit: int = 0) -> TagType:
         """Traverse the tree and list the children of a given tag
 
         Args:
             tag: The node tag
             parent: The ObjectID of the node's parent
+            document_limit: Maximum number of documents to return. If set to zero there is no limit
 
         Returns:
             A list of names of the children
         """
-
         if len(tag) == 0:
             return list(map(itemgetter('tag'),
-                            self._collection.find({'parent': parent, 'tag': {'$exists': True}}, {'value': 0})))
+                            self._collection.find({'parent': parent, 'tag': {'$exists': True}},
+                                                  {'value': 0},
+                                                  limit=document_limit,
+                                                  sort=[('tag', -1)])))
```

Comment:
@peendebak would it be better to add a sort_field and sort_order parameter to _retrieve_nodes_by_tag ( ) rather than always doing a descending sort on tag field?

Reply:
Not needed when there is a good reason for descending, because other method are depending on this sort order it will be complex to make it parameterized.